src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1546,16 +1549,38 @@
        new_value, old_value, message, file, line, containment_path, containing_class, corrective_change
      from resource_events"]
    (fn [rows]
-     (reduce (fn [hashes-seen row]
-               (conj hashes-seen
-                     (let [hash-str (hash/resource-event-identity-pkey row)]
-                       (when-not (hashes-seen hash-str)
-                         (jdbc/insert-multi! "resource_events_transform"
-                                             (list (assoc row :event_hash
-                                                          (sutils/munge-hash-for-storage hash-str)))))
-                       hash-str)))
-             #{}
-             rows)))
+     (let [old-cols '(:report_id :certname_id :status :timestamp :resource_type
+                      :resource_title :property :new_value :old_value :message
+                      :file :line :containment_path :containing_class
+                      :corrective_change)
+           new-cols (conj old-cols :event_hash)
+           update-row (apply juxt (conj old-cols
+                                        (comp sutils/munge-hash-for-storage
+                                              hash/resource-event-identity-pkey)))
```

Comment:
If you'd rather leave it alone, no worries, but I'd be tempted to change old-cols to a vector, create new-cols as `(apply vector :event_hash old-cols)` (or could just use cons, if it doesn't need to be a vector), and then use `(apply juxt (comp ...) old-cols)`.  That'd be more conventional, but more notably, clojure-mode won't try to indent it wrong every time paredit, etc. adjusts the form, and it avoids people having to remember conj's "direction-sensitivity".

Reply:
Switched to a vector.